sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -79,6 +81,28 @@ public DirectConnectionConfig setConnectTimeout(Duration connectTimeout) {
         return this;
     }
 
+    /**
+     * Gets a value that indicates whether Direct TCP connection endpoint rediscovery should is enabled.
```

Comment:
Expand on what it is from CX perspective. 

QQ; Do we want this feature flag as prominent in ConnectionConfig?

Reply:
It is a good question whether we want this flag as prominent as it is. This PR now enables connection endpoint rediscovery by default. I would prefer not to advertise the feature until we've got more experience with it. Enabling the feature by default is counter to that.

@kushagraThapar, @moderakh what do you think?

Reply:
IMO, any feature may have bug on its first release. I prefer if we change the default to enabled later when we are more confident on the behaviour.